script/fixes/105_modaction_to_postevents.rb

```diff
@@ -2,8 +2,24 @@
 
 require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'config', 'environment'))
 
+class PostEventTemp < PostEvent
+  self.table_name = "post_events_temp"
+end
+
+ActiveRecord::Base.connection.execute('CREATE TABLE post_events_temp (
+	id bigserial NOT NULL,
+	creator_id int8 NOT NULL,
+	post_id int8 NOT NULL,
+	"action" int4 NOT NULL,
+	extra_data jsonb NOT NULL,
+	created_at timestamp NOT NULL,
+	CONSTRAINT post_events_temp_pkey PRIMARY KEY (id)
+);')
+
+ActiveRecord::Base.connection.execute('CREATE INDEX index_post_events_temp_on_created_at ON public.post_events_temp USING btree (created_at);')
+
 def create(post_id, creator_id, created_at, action, extra_data = {})
-  PostEvent.create!(post_id: post_id, creator_id: creator_id, created_at: created_at, action: action, extra_data: extra_data)
+  PostEventTemp.create!(post_id: post_id, creator_id: creator_id, created_at: created_at, action: action, extra_data: extra_data)
 end
 
 PostApproval.in_batches do |batch|
```

Comment:
This should likely be `find_in_batches` otherwise it will emit multiple queries per batch. the `find_in_batches` variant calls `in_batches` with load set to true so that it doesn't return an ActiveRecord::Relationship, but an array of results.

Reply:
My bad, changed it to `find_in_batches`. After a quick google search it seems that in_batches runs two queries per batch.